### PR TITLE
feat(service-portal): Add simple path masking

### PR DESCRIPTION
**Pull request by thordurhhh**
## What

Add simple path masking. Not really to be used for top security, but fine for low level.

[Feature deployment available](https://service-portalnat-id-path-masking-beta.dev01.devland.is/minarsidur) ✨ 

## Why

So we won't have any national Id's in paths. A simple base64 string that's not reversible.

## Screenshots / Gifs

Screenshot 2023-11-21 at 14 24 27

This should not slow anything down. Here's a screenshot of the performance test

Once:
Screenshot 2023-11-21 at 14 23 04

1000 iterations:
Screenshot 2023-11-21 at 14 20 55

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/shared/utils/src/lib/simpleEncryption.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+const ALGORITHM = 'aes-256-cbc'
+
+/**
+ *
+ * @param text The string you wish to hide
+ * @param key You secret key
+ * @returns URL safe Base64
+ */
+export const encrypt = (text: string, key: string): string => {
```
Comment (osk) [inline libs/shared/utils/src/lib/simpleEncryption.ts:12]: Maybe we should call this something other than `encrypt`? `maskString`, `maskData`?

File: libs/shared/utils/src/lib/simpleEncryption.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  const derivedKey = hashKey(key)
+
+  const cipher = createCipheriv(ALGORITHM, derivedKey, Buffer.alloc(16))
+  const encrypted =
+    cipher.update(text, 'utf-8', 'base64') + cipher.final('base64')
+
+  return Base64.encodeURI(encrypted)
```
Comment (osk) [inline libs/shared/utils/src/lib/simpleEncryption.ts:19]: Could this throw? Should we handle here and return `string | null`?

File: libs/shared/utils/src/lib/simpleEncryption.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    const encryptedData = Base64.decode(encryptedText)
+    const derivedKey = hashKey(key)
+    const decipher = createDecipheriv(ALGORITHM, derivedKey, Buffer.alloc(16))
+
+    return (
+      decipher.update(encryptedData, 'base64', 'utf-8') +
+      decipher.final('utf-8')
+    )
+  } catch (e) {
+    return encryptedText
```
Comment (osk) [inline libs/shared/utils/src/lib/simpleEncryption.ts:38]: Log error?
Also returning the `encryptedText` could cause weird behaviour, I'd rather return `null`

File: libs/service-portal/information/src/screens/Child/Child.tsx [hunk 4]
```diff
@@ -29,21 +29,22 @@ import { spmm } from '../../lib/messages'
...
   const { formatMessage } = useLocale()
   const userInfo = useUserInfo()
-  const { nationalId } = useParams() as UseParams
+  const { baseId } = useParams() as UseParams
 
   const { data, loading, error } = useNationalRegistryChildCustodyQuery({
     variables: {
       api: 'v3',
-      childNationalId: nationalId,
+      childNationalId: decrypt(baseId, userInfo.profile.nationalId),
```
Comment (osk) [inline libs/service-portal/information/src/screens/Child/Child.tsx:47]: Need to handle if `decrypt` fails

File: libs/service-portal/information/src/screens/Child/Child.tsx [hunk 5]
```diff
@@ -72,7 +74,7 @@ const Child = () => {
   )
 
   const isChildOrChildOfUser = isChild || isChildOfUser
-  if (!nationalId || (!loading && !isChildOrChildOfUser && !error))
+  if (!baseId || (!loading && !isChildOrChildOfUser && !error))
```
Comment (osk) [inline libs/service-portal/information/src/screens/Child/Child.tsx:77]: This makes my head spin a bit (but that is for another time to fix)

File: libs/service-portal/information/src/components/FamilyMemberCard/FamilyMemberCard.tsx [hunk 6]
```diff
@@ -8,6 +8,7 @@ import { InformationPaths } from '../../lib/paths'
 interface Props {
   title: string
   nationalId: string
+  baseId?: string
```
Comment (osk) [inline libs/service-portal/information/src/components/FamilyMemberCard/FamilyMemberCard.tsx:11]: What does it mean if `baseId` is `undefined`?

Comment (thordurhhh) [reply libs/service-portal/information/src/components/FamilyMemberCard/FamilyMemberCard.tsx:11]: Not all every familymember has a baseId. Like for example the logged in user and spouse.

**Review by thorkellmani — APPROVED** [review]
LGTM 🔎

**Pull request merged**